File: room-manager/src/main/java/org/rivero/roommanagement/services/ReceiptConsumerService.java
package org.rivero.roommanagement.services;

import lombok.RequiredArgsConstructor;
import org.rivero.roommanagement.entities.ReceiptConsumer;
import org.rivero.roommanagement.repositories.DBConnectionManager;
import org.rivero.roommanagement.repositories.ReceiptConsumerRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ReceiptConsumerService {
    private final ReceiptConsumerRepository receiptConsumerRepository;
    final DBConnectionManager dbConnectionManager;

    public void create(ReceiptConsumer request) {
        receiptConsumerRepository.insert(request);
    }

    public List<ReceiptConsumer> getByUserId(String id) {
        return receiptConsumerRepository.getListByUserId(id);
    }

    public List<ReceiptConsumer> getByReceiptId(String id) {
        return receiptConsumerRepository.getListByReceiptId(id);
    }

    public String deleteOne(String id) {
        receiptConsumerRepository.deleteOne(id);
        return "Record deleted";
    }
}

File: room-manager/src/main/kotlin/org/rivero/roommanager/subscription/ReceiptConsumerService.kt
package org.rivero.roommanager.subscription

import lombok.RequiredArgsConstructor
import org.rivero.roommanager.DBConnectionManager
import org.rivero.roommanager.entities.ReceiptConsumer
import org.rivero.roommanager.receipt.ReceiptConsumerRepository
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class ReceiptConsumerService {
    private val receiptConsumerRepository: ReceiptConsumerRepository? = null
    val dbConnectionManager: DBConnectionManager? = null

    fun create(request: ReceiptConsumer?) {
        receiptConsumerRepository!!.insert(request!!)
    }

    fun getByUserId(id: String?): List<ReceiptConsumer> {
        return receiptConsumerRepository!!.getListByUserId(id)
    }

    fun getByReceiptId(id: String?): List<ReceiptConsumer> {
        return receiptConsumerRepository!!.getListByReceiptId(id)
    }

    fun deleteOne(id: String?): String {
        receiptConsumerRepository!!.deleteOne(id)
        return "Record deleted"
    }
}
